### ISSUE TITLE
`mkdocs-material` annotations support

### ISSUE THREAD

**Issue by Viicos** [provinzkraut/AutoPyTabs#7]
Thanks for creating this extension. At Pydantic, we have a similar [custom implementation](https://github.com/pydantic/pydantic/blob/95c374da590c6da9468ae29dcd2a93f49a560654/docs/plugins/main.py#L133) to create tabs, but it has been a source of bugs and I wanted to move away from this approach.

Unfortunately, it seems like the markdown implementation has to be done via regex, and as you probably experienced the Sphinx/docutils approach is easier to reason about.

I want to switch from our current implementation to this extension. However, we make use of [`mkdocs-material` annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations). Looking at the implementation of annotations, it seems to be a `mkdocs-material` specific feature [implemented in JS](https://github.com/squidfunk/mkdocs-material/blob/f947bbe64a6f0f4754c0a7a1afc3b67e8b735c8b/src/templates/assets/javascripts/components/content/annotation/list/index.ts), and what it does is it replaces lists with annotations:

``````markdown
```python
a = 1  # (1)!
```

1. I will be replaced when the page is rendered
``````

This is imo a poor design choice, but there isn't much we can do.

Do you think the regex logic of the `markdown_ext.py` module could be updated to support annotations? Here are some valid examples, for reference:

``````markdown
```python
a = 1  # (1)!
b = 2  # (2)!
```


1. An arbitrary amount of new lines can be used between the code block and the first element of the list.


2. Same for any following list element.
I can also span multiple lines.

=== "Tab 1"

    ```python
    a: int  # (1)!
    ```

    1. Annotation in tab. Watch out for the indentation.

=== "Tab 2"

    ```python
    a: int
    ```
``````

Note that the [new block API](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/tab/) could be used, as this avoids headaches with indentation.

**Comment by provinzkraut** [comment]
> Do you think the regex logic of the `markdown_ext.py` module could be updated to support annotations? 

Yes, definitely. However, I'll say that I currently don't have much time to sink into this, so I won't be working on it in the near future. PRs are welcome though.

<hr>

I haven't actually tried to use the markdown extensions with annotations yet. What exactly is the issue there?
